### ISSUE TITLE
[Feature request] Ability to switch back to previous layout

### ISSUE THREAD

**Issue by wincent** [tmux/tmux#859]
Sometimes I have a bunch of panes in a window that I have configured manually and I accidentally hit `[prefix] Space`, which causes tmux to jump to the next preset layout, destroying my set-up. 

(This happens more often than I would like because I'm using `^Space` as my prefix, and I've also bound `[prefix] ^Space` to `last-window`, seeing as I use it so often: `^Space ^Space` and `^Space Space` are just too similar, so I generally screw this up at least once a day.)

> The feature request: **In cases like this, I would love a binding to be able to go back to the immediately prior layout.**

Workarounds for now:

- If I've spent a bunch of time making a useful layout, manually `tmux list-windows` to record it so that I can reapply it with `tmux select-layout`. (I am too forgetful for this.)
- If you forget to `list-windows`, painstakingly manually construct a layout description in the format required by `select-layout` and apply it. (I am too impatient for this.)
- Rebind `[prefix] Space` to make it less easy to hit by mistake. (I will probably do this.)

So, this is partly my own fault and I have at least one viable workaround for it, but I thought I'd throw up the feature request anyway as it may be interesting to others.

**Comment by nicm** [comment]
There is already a way to do this, I can't remember the command, look at
select-layout





On 14 Apr 2017 11:05 pm, "Greg Hurrell" <notifications@github.com> wrote:

Sometimes I have a bunch of panes in a window that I have configured
manually and I accidentally hit [prefix] Space, which causes tmux to jump
to the next preset layout, destroying my set-up.

(This happens more often than I would like because I'm using ^Space as my
prefix, and I've also bound [prefix] ^Space to last-window, seeing as I use
it so often: ^Space ^Space and ^Space Space are just too similar, so I
generally screw this up at least once a day.)

The feature request: *In cases like this, I would love a binding to be able
to go back to the immediately prior layout.*

Workarounds for now:

   - If I've spent a bunch of time making a useful layout, manually tmux
   list-windows to record it so that I can reapply it with tmux
   select-layout. (I am too forgetful for this.)
   - If you forget to list-windows, painstakingly manually construct a
   layout description in the format required by select-layout and apply it.
   (I am too impatient for this.)
   - Rebind [prefix] Space to make it less easy to hit by mistake. (I will
   probably do this.)

So, this is partly my own fault and I have at least one viable workaround
for it, but I thought I'd throw up the feature request anyway as it may be
interesting to others.

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/tmux/tmux/issues/859>, or mute the thread
